src/VisualStudio/FSharp/Microsoft.VisualStudio.LanguageServices.FSharp.csproj

```diff
@@ -0,0 +1,30 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright (c)  Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information. -->
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <Platform Condition="'$(Platform)' == ''">AnyCPU</Platform>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <OutputType>Library</OutputType>
+    <RootNamespace>Microsoft.VisualStudio.LanguageServices.FSharp</RootNamespace>
+    <TargetFramework>net472</TargetFramework>
+
+    <!-- NuGet -->
+    <IsPackable>true</IsPackable>
+    <PackageDescription>
+      A private package for the FSharp team to grant access to certain internal APIs.
+    </PackageDescription>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Release|AnyCPU'" />
+  <ItemGroup>
+    <Reference Include="System" />
+    <Reference Include="System.Core" />
+  </ItemGroup>
+  <ItemGroup>
+    <ProjectReference Include="..\..\Compilers\Core\Portable\Microsoft.CodeAnalysis.csproj" />
+    <ProjectReference Include="..\..\Workspaces\Core\Portable\Microsoft.CodeAnalysis.Workspaces.csproj" />
+  </ItemGroup>
+  <ItemGroup>
+    <InternalsVisibleTo Include="FSharp.Editor" Key="$(FSharpKey)" />
```

Comment:
❗️ This will allow APIs from other layers to be exported in the "friend API" of this layer, without any compiler protection. The design in #31393 relies on the adapter assembly *not* providing IVT access to other assemblies.

While exposing an API in this layer as `public` would allow any other extension to call the API, this action would not be supported, and the API would not be packaged in a location like NuGet where it's easy to assume that the API has shipped as stable. If a third-party extension breaks because they incorrectly referenced a private assembly that later changed, then the third-party extension breaks. This is normal versioning behavior for this type of assembly, and we should not have reservations about removing this `InternalsVisibleTo` line. It's not like we'll be any more exposed [than we already are](https://github.com/tunnelvisionlabs/InheritanceMargin/blob/fbd49f9d51f91da4567145d79cb2f6c6e3dd26d9/Tvl.VisualStudio.InheritanceMargin.CSharp/CSharpInheritanceAnalyzer.cs#L65-L90).

Reply:
➡️ This approach makes things more difficult for downstream consumers, but not more difficult for Roslyn developers. I'm fine with deferring it until such time as the consumers of the adapter assemblies raise a complaint.